assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py

```diff
@@ -442,6 +444,121 @@ def copy_preprocess_args(args: Namespace) -> Namespace:
     return args
 
 
+def identify_deepspeed_stage(deepspeed_config: str) -> int:
+    """Read the deepspeed stage from the deepspeed config."""
+    with open(deepspeed_config, "r", encoding="utf-8") as fp:
+        ds_config_json = json.load(fp)
+    zero_optimization_config = ds_config_json.get("zero_optimization", {})
+
+    ds_stage = zero_optimization_config.get("stage", None)
+    if not isinstance(ds_stage, int):
+        raise ACFTValidationException._with_error(
+                AzureMLError.create(
+                    ACFTUserError,
+                    pii_safe_message=(
+                        "Invalid deepspeed config file. Stage information is missing from the config file."
+                    )
+                )
+            )
+
+    return ds_stage
+
+
+def check_for_invalid_ds_zero3_settings(args: Namespace):
+    """Check if invalid ds3 settings are selected by the user.
+
+    If fail_run is enabled for a setting raise an User Error otherwise reset the args using the valid_settings.
+    :param args: User passed args
+    :type Namespace
+    """
+    invalid_ds_zero3_settings = [
+        dict(
+            invalid_settings=dict(apply_lora=True),
+            fail_run=True,
+            valid_settings=None
+        ),
+        dict(
+            invalid_settings=dict(apply_ort=True),
+            fail_run=True,
+            valid_settings=None
+        ),
```

Comment:
Like for auto_find_batch_size shouldn't we setting the values to false?

Reply:
There are 2 types of validations one that fails the run and 1 that resets the value. _apply_ort_ and _apply_lora_ are set with fail_run=True i.e. unless these settings are enabled, we are going to fail the run. On the other hand, with _auto_find_batch_size=True_, fail_run=False. Here we need the valid configurations to reset the settings.